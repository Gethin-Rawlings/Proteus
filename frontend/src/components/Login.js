import React from 'react';
import "./login.css";
import 'whatwg-fetch';
import Footer from "./Footer"
import Welcome from "./Welcome"
import { login } from './apiCalls'

class Login extends React.Component {
    constructor(props) { 
            super(props); 
              this.state={value: ''};
              this.handleChange = this.handleChange.bind(this);
              this.handleSubmit = this.handleSubmit.bind(this);      
           } 
           handleChange(event) {
            const target = event.target;
            const name = target.name;
            const value = target.value;
            this.setState({[name]: value});
          }
           handleSubmit(event) { 
            event.preventDefault(); 
            const { history } = this.props;
            const data = new FormData(event.target);
            try {
              const data = login(data)
              sessionStorage.setItem('token',data.token);
              sessionStorage.setItem('loggedIn',data.success);
              sessionStorage.setItem('supplier',data.supplier)
              sessionStorage.setItem('network',data.network)
              sessionStorage.setItem('admin',data.admin)
              sessionStorage.setItem('finance',data.finance)
              sessionStorage.setItem('commission',data.commission)
              history.push("/main")
            } catch(err) {
              console.log("Login Failed")
            } 
           } 
           render() { 
             return ( 
              <div className="App">
                <Welcome />
                <section className="App-intro">
                <form className='content' onSubmit={this.handleSubmit} id='login'> 
                    <input  name="username" className="username" type="text"  placeholder="username" value={this.state.username} onChange={this.handleChange}/> 
                    <input  name="password" className="password" type="password"  placeholder="password" value={this.state.password} onChange={this.handleChange}/> 
                    <button id="loginbutton" className="loginbutton">Login</button> 
                 <section className="browser">Recomended browsers are Chrome, Safari or Firefox</section>
                </form> 
               </section>
               <Footer />
               </div>
             ); 
           } 
         } 

  export default Login;

 
