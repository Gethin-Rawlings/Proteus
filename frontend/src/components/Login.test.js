import React from 'react';
import * as Enzyme from 'enzyme';
import { shallow, mount } from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import Login from './Login'
import ReactDOM from 'react-dom';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

test('Login component should render as expected', () => {
    const component = shallow(<Login />)
    const tree = component;
    expect(tree).toMatchSnapshot();
})

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Login />, div);
});

test('Login component should render as expected', () => {
    const component = mount(<Login />)
    expect(component.find(Login).length).toBe(1)
})
