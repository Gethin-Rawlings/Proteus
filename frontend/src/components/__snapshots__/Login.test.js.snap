// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Footer />,
        <section
          className="App-intro"
>
          <form
                    className="content"
                    id="login"
                    onSubmit={[Function]}
          >
                    <input
                              className="username"
                              name="username"
                              onChange={[Function]}
                              placeholder="username"
                              type="text"
                              value={undefined}
                    />
                    <input
                              className="password"
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="password"
                              value={undefined}
                    />
                    <button
                              className="loginbutton"
                              id="loginbutton"
                    >
                              Login
                    </button>
                    <section
                              className="browser"
                    >
                              Recomended browsers are Chrome, Safari or Firefox
                    </section>
          </form>
</section>,
        <Footer />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="content"
            id="login"
            onSubmit={[Function]}
>
            <input
                        className="username"
                        name="username"
                        onChange={[Function]}
                        placeholder="username"
                        type="text"
                        value={undefined}
            />
            <input
                        className="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value={undefined}
            />
            <button
                        className="loginbutton"
                        id="loginbutton"
            >
                        Login
            </button>
            <section
                        className="browser"
            >
                        Recomended browsers are Chrome, Safari or Firefox
            </section>
</form>,
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="username"
                name="username"
                onChange={[Function]}
                placeholder="username"
                type="text"
                value={undefined}
/>,
              <input
                className="password"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value={undefined}
/>,
              <button
                className="loginbutton"
                id="loginbutton"
>
                Login
</button>,
              <section
                className="browser"
>
                Recomended browsers are Chrome, Safari or Firefox
</section>,
            ],
            "className": "content",
            "id": "login",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "username",
                "name": "username",
                "onChange": [Function],
                "placeholder": "username",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "loginbutton",
                "id": "loginbutton",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recomended browsers are Chrome, Safari or Firefox",
                "className": "browser",
              },
              "ref": null,
              "rendered": "Recomended browsers are Chrome, Safari or Firefox",
              "type": "section",
            },
          ],
          "type": "form",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Footer />,
          <section
            className="App-intro"
>
            <form
                        className="content"
                        id="login"
                        onSubmit={[Function]}
            >
                        <input
                                    className="username"
                                    name="username"
                                    onChange={[Function]}
                                    placeholder="username"
                                    type="text"
                                    value={undefined}
                        />
                        <input
                                    className="password"
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="password"
                                    type="password"
                                    value={undefined}
                        />
                        <button
                                    className="loginbutton"
                                    id="loginbutton"
                        >
                                    Login
                        </button>
                        <section
                                    className="browser"
                        >
                                    Recomended browsers are Chrome, Safari or Firefox
                        </section>
            </form>
</section>,
          <Footer />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="content"
              id="login"
              onSubmit={[Function]}
>
              <input
                            className="username"
                            name="username"
                            onChange={[Function]}
                            placeholder="username"
                            type="text"
                            value={undefined}
              />
              <input
                            className="password"
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="password"
                            value={undefined}
              />
              <button
                            className="loginbutton"
                            id="loginbutton"
              >
                            Login
              </button>
              <section
                            className="browser"
              >
                            Recomended browsers are Chrome, Safari or Firefox
              </section>
</form>,
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value={undefined}
/>,
                <input
                  className="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value={undefined}
/>,
                <button
                  className="loginbutton"
                  id="loginbutton"
>
                  Login
</button>,
                <section
                  className="browser"
>
                  Recomended browsers are Chrome, Safari or Firefox
</section>,
              ],
              "className": "content",
              "id": "login",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "username",
                  "name": "username",
                  "onChange": [Function],
                  "placeholder": "username",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "password",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "loginbutton",
                  "id": "loginbutton",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recomended browsers are Chrome, Safari or Firefox",
                  "className": "browser",
                },
                "ref": null,
                "rendered": "Recomended browsers are Chrome, Safari or Firefox",
                "type": "section",
              },
            ],
            "type": "form",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
